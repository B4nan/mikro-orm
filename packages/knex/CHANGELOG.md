# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)


### Performance Improvements

* **core:** do not use `em.merge()` internally ([6a1a6d6](https://github.com/mikro-orm/mikro-orm/commit/6a1a6d68b65a20b8f1a78bf644844427f3b2dd1a))





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/knex





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)


### Bug Fixes

* **core:** refactor internals to reduce number of cycles ([#830](https://github.com/mikro-orm/mikro-orm/issues/830)) ([3994767](https://github.com/mikro-orm/mikro-orm/commit/3994767d93ef119d229bedffa77eb2ea3af5c775))
