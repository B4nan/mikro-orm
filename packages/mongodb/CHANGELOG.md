# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.2.3](https://github.com/mikro-orm/mikro-orm/compare/v4.2.2...v4.2.3) (2020-10-24)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.2.2](https://github.com/mikro-orm/mikro-orm/compare/v4.2.1...v4.2.2) (2020-10-22)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.2.1](https://github.com/mikro-orm/mikro-orm/compare/v4.2.0...v4.2.1) (2020-10-20)

**Note:** Version bump only for package @mikro-orm/mongodb





# [4.2.0](https://github.com/mikro-orm/mikro-orm/compare/v4.1.1...v4.2.0) (2020-10-18)


### Bug Fixes

* **core:** update umzug types to 2.3 ([4668e78](https://github.com/mikro-orm/mikro-orm/commit/4668e78d1c900d1718625364b603dda1b707dd82)), closes [#926](https://github.com/mikro-orm/mikro-orm/issues/926)


### Features

* **core:** allow defining multiple entities in single file ([e3ab336](https://github.com/mikro-orm/mikro-orm/commit/e3ab33699b116e25e09a7449589db7955899c8ac)), closes [#922](https://github.com/mikro-orm/mikro-orm/issues/922)
* **core:** allow storing embeddables as objects ([#927](https://github.com/mikro-orm/mikro-orm/issues/927)) ([ba881e6](https://github.com/mikro-orm/mikro-orm/commit/ba881e6257dd5d72bb10ca402b0322f7dbbda69c)), closes [#906](https://github.com/mikro-orm/mikro-orm/issues/906)





# [4.1.0](https://github.com/mikro-orm/mikro-orm/compare/v4.0.7...v4.1.0) (2020-10-12)


### Bug Fixes

* **mongo:** filter by serialized PK inside group condition ([a492a64](https://github.com/mikro-orm/mikro-orm/commit/a492a64e08fc4fac5b447d0928bbc6d7b453e29f)), closes [#908](https://github.com/mikro-orm/mikro-orm/issues/908)
* **typings:** improve inference of the entity type ([67f8015](https://github.com/mikro-orm/mikro-orm/commit/67f80157ae013479b6fc47ae1c08a5cd31a6c32d)), closes [#876](https://github.com/mikro-orm/mikro-orm/issues/876)


### Performance Improvements

* **core:** implement bulk updates in mongo driver ([5f347c1](https://github.com/mikro-orm/mikro-orm/commit/5f347c1de4e5dd6f30305275c86d333611edc27c)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** implement bulk updates in sql drivers ([b005353](https://github.com/mikro-orm/mikro-orm/commit/b00535349368ba18a5e0a5452ae7e2b567e12952)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** optimize entity hydration ([6c56a05](https://github.com/mikro-orm/mikro-orm/commit/6c56a05a86b78fc9c3ebc6ddceb75072289e6b48)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use bulk inserts in all drivers ([10f2e55](https://github.com/mikro-orm/mikro-orm/commit/10f2e55f6710822c9fca272b01c278ff80065096)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use dedicated identity maps for each entity ([84667f9](https://github.com/mikro-orm/mikro-orm/commit/84667f9e97323e4b054db2c0c70939ab0ca86c86)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use faster way to check number of object keys ([82f3ee4](https://github.com/mikro-orm/mikro-orm/commit/82f3ee4d4169def8ce8fe31764171193e8b8b5dc)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use JIT compiled PK getters/serializers ([0ec99dc](https://github.com/mikro-orm/mikro-orm/commit/0ec99dc75690bf15df3897b4da0fc3b2ab709cdd)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)





## [4.0.7](https://github.com/mikro-orm/mikro-orm/compare/v4.0.6...v4.0.7) (2020-09-24)


### Features

* **core:** add groupBy, having and schema to `CountOptions` ([d3c3858](https://github.com/mikro-orm/mikro-orm/commit/d3c38584c38e11002460a6556405e136aabefa93))


### Performance Improvements

* **core:** use batch inserts in UoW (postgres & mongodb) ([#865](https://github.com/mikro-orm/mikro-orm/issues/865)) ([54ad928](https://github.com/mikro-orm/mikro-orm/commit/54ad928aab44d0c42e3f7b306eef0c07ed65dfc1)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)





## [4.0.6](https://github.com/mikro-orm/mikro-orm/compare/v4.0.5...v4.0.6) (2020-09-22)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.5](https://github.com/mikro-orm/mikro-orm/compare/v4.0.4...v4.0.5) (2020-09-21)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.4](https://github.com/mikro-orm/mikro-orm/compare/v4.0.3...v4.0.4) (2020-09-19)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)

**Note:** Version bump only for package @mikro-orm/mongodb
