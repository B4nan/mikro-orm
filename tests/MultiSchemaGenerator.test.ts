import { BASE_DIR, initORMMySql, initORMPostgreSql, initORMSqlite, initORMSqlite2 } from './bootstrap';
import { EntitySchema, ReferenceType, Utils, MikroORM } from '@mikro-orm/core';
import { SchemaGenerator, EntityManager } from '@mikro-orm/knex';
import { FooBarSchema2, FooBazSchema2 } from './entities-with-schema';
import { MongoDriver, MongoEntityManager } from '@mikro-orm/mongodb';

describe('SchemaGenerator', () => {

  test('create/drop multi-schema databases [mysql]', async () => {
    const dbName = `mikro_orm_test_${Date.now()}`;
    const orm = await MikroORM.init({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName,
      port: 3307,
      baseDir: BASE_DIR,
      type: 'mysql',
    });

    const generator = new SchemaGenerator(orm.em as EntityManager);
    await generator.ensureDatabase();
    await generator.dropAllSchemas();
    await orm.close(true);

    return true;
  });

  test('generate multi-schema database from metadata [mysql]', async () => {
    const orm = await MikroORM.init({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'mikro_orm_test',
      port: 3307,
      baseDir: BASE_DIR,
      type: 'mysql',
      migrations: { path: BASE_DIR + '/../temp/migrations' },
    });
    const generator = new SchemaGenerator(orm.em as EntityManager);
    await generator.dropAllSchemas();
    const dump = await generator.generate();
    expect(dump).toMatchSnapshot('mysql-schema-dump');

    const dropDump = await generator.getDropSchemaSQL();
    expect(dropDump).toMatchSnapshot('mysql-drop-schema-dump');

    const createDump = await generator.getCreateSchemaSQL();
    expect(createDump).toMatchSnapshot('mysql-create-schema-dump');

    const updateDump = await generator.getUpdateSchemaSQL();
    expect(updateDump).toMatchSnapshot('mysql-update-schema-dump');

    await orm.close(true);
  });

  test('generate multi-schema database from metadata [mariadb]', async () => {
    const orm = await MikroORM.init({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'mikro_orm_test',
      port: 3309,
      baseDir: BASE_DIR,
      type: 'mariadb',
      migrations: { path: BASE_DIR + '/../temp/migrations' },
    });
    const generator = new SchemaGenerator(orm.em as EntityManager);
    await generator.dropAllSchemas();
    const dump = await generator.generate();
    expect(dump).toMatchSnapshot('mariadb-schema-dump');

    const dropDump = await generator.getDropSchemaSQL();
    expect(dropDump).toMatchSnapshot('mariadb-drop-schema-dump');

    const createDump = await generator.getCreateSchemaSQL();
    expect(createDump).toMatchSnapshot('mariadb-create-schema-dump');

    const updateDump = await generator.getUpdateSchemaSQL();
    expect(updateDump).toMatchSnapshot('mariadb-update-schema-dump');

    await orm.close(true);
  });

  test('generate multi-schema database from metadata [postgresql]', async () => {
    const orm = await MikroORM.init({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'postgres',
      port: 5432,
      baseDir: BASE_DIR,
      type: 'postgresql',
      migrations: { path: BASE_DIR + '/../temp/migrations' },
    });
    const generator = new SchemaGenerator(orm.em as EntityManager);
    await generator.dropAllSchemas();
    const dump = await generator.generate();
    expect(dump).toMatchSnapshot('postgresql-schema-dump');

    const dropDump = await generator.getDropSchemaSQL();
    expect(dropDump).toMatchSnapshot('postgresql-drop-schema-dump');

    const createDump = await generator.getCreateSchemaSQL();
    expect(createDump).toMatchSnapshot('postgresql-create-schema-dump');

    const updateDump = await generator.getUpdateSchemaSQL();
    expect(updateDump).toMatchSnapshot('postgresql-update-schema-dump');

    await orm.close(true);
  });
});
