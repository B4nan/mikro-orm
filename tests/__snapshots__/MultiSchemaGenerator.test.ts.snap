// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaGenerator generate multi-schema database from metadata [mariadb]: mariadb-create-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mariadb]: mariadb-drop-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

drop table if exists \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\`;
drop table if exists \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\`;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mariadb]: mariadb-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

drop table if exists \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\`;
drop table if exists \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\`;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mariadb]: mariadb-update-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mysql]: mysql-create-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mysql]: mysql-drop-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

drop table if exists \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\`;
drop table if exists \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\`;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mysql]: mysql-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

drop table if exists \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\`;
drop table if exists \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\`;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [mysql]: mysql-update-schema-dump 1`] = `
"set names utf8mb4;
set foreign_key_checks = 0;

create table \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`version\` datetime(3) not null default current_timestamp(3)) default character set utf8mb4 engine = InnoDB;

create table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\` int unsigned not null auto_increment primary key, \`name\` varchar(255) not null, \`baz_id\` int unsigned null, \`foo_bar_id\` int unsigned null, \`version\` datetime not null default current_timestamp, \`blob\` blob null, \`array\` text null, \`object\` json null) default character set utf8mb4 engine = InnoDB;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_baz_id_index\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_baz_id_unique\`(\`baz_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add index \`foo_bar_schema2_foo_bar_id_index\`(\`foo_bar_id\`);
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add unique \`foo_bar_schema2_foo_bar_id_unique\`(\`foo_bar_id\`);

alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_baz_id_foreign\` foreign key (\`baz_id\`) references \`mikro_orm_test_multi_2\`.\`foo_baz_schema2\` (\`id\`) on update cascade on delete cascade;
alter table \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` add constraint \`foo_bar_schema2_foo_bar_id_foreign\` foreign key (\`foo_bar_id\`) references \`mikro_orm_test_multi_1\`.\`foo_bar_schema2\` (\`id\`) on update cascade on delete set null;

set foreign_key_checks = 1;
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [postgresql]: postgresql-create-schema-dump 1`] = `
"set names 'utf8';
set session_replication_role = 'replica';

create schema if not exists \\"mikro_orm_test_multi_2\\";

create schema if not exists \\"mikro_orm_test_multi_1\\";

create table \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"version\\" timestamptz(3) not null default current_timestamp(3));

create table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"baz_id\\" int null, \\"foo_bar_id\\" int null, \\"version\\" timestamptz(0) not null default current_timestamp(0), \\"blob\\" bytea null, \\"array\\" text[] null, \\"object\\" jsonb null);
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_unique\\" unique (\\"baz_id\\");
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_unique\\" unique (\\"foo_bar_id\\");

alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_foreign\\" foreign key (\\"baz_id\\") references \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\") on update cascade on delete cascade;
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_foreign\\" foreign key (\\"foo_bar_id\\") references \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\") on update cascade on delete set null;

set session_replication_role = 'origin';
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [postgresql]: postgresql-drop-schema-dump 1`] = `
"set names 'utf8';
set session_replication_role = 'replica';

drop table if exists \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" cascade;
drop table if exists \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" cascade;

set session_replication_role = 'origin';
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [postgresql]: postgresql-schema-dump 1`] = `
"set names 'utf8';
set session_replication_role = 'replica';

drop table if exists \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" cascade;
drop table if exists \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" cascade;

create schema if not exists \\"mikro_orm_test_multi_2\\";

create schema if not exists \\"mikro_orm_test_multi_1\\";

create table \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"version\\" timestamptz(3) not null default current_timestamp(3));

create table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"baz_id\\" int null, \\"foo_bar_id\\" int null, \\"version\\" timestamptz(0) not null default current_timestamp(0), \\"blob\\" bytea null, \\"array\\" text[] null, \\"object\\" jsonb null);
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_unique\\" unique (\\"baz_id\\");
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_unique\\" unique (\\"foo_bar_id\\");

alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_foreign\\" foreign key (\\"baz_id\\") references \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\") on update cascade on delete cascade;
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_foreign\\" foreign key (\\"foo_bar_id\\") references \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\") on update cascade on delete set null;

set session_replication_role = 'origin';
"
`;

exports[`SchemaGenerator generate multi-schema database from metadata [postgresql]: postgresql-update-schema-dump 1`] = `
"set names 'utf8';
set session_replication_role = 'replica';

create schema \\"mikro_orm_test_multi_2\\";

create schema \\"mikro_orm_test_multi_1\\";

create table \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"version\\" timestamptz(3) not null default current_timestamp(3));

create table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\" serial primary key, \\"name\\" varchar(255) not null, \\"baz_id\\" int null, \\"foo_bar_id\\" int null, \\"version\\" timestamptz(0) not null default current_timestamp(0), \\"blob\\" bytea null, \\"array\\" text[] null, \\"object\\" jsonb null);
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_unique\\" unique (\\"baz_id\\");
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_unique\\" unique (\\"foo_bar_id\\");

alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_baz_id_foreign\\" foreign key (\\"baz_id\\") references \\"mikro_orm_test_multi_2\\".\\"foo_baz_schema2\\" (\\"id\\") on update cascade on delete cascade;
alter table \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" add constraint \\"foo_bar_schema2_foo_bar_id_foreign\\" foreign key (\\"foo_bar_id\\") references \\"mikro_orm_test_multi_1\\".\\"foo_bar_schema2\\" (\\"id\\") on update cascade on delete set null;

set session_replication_role = 'origin';
"
`;
